crates/bevy_reflect/src/struct_trait.rs

```diff
@@ -508,6 +508,44 @@ impl Debug for DynamicStruct {
     }
 }
 
+impl FromIterator<(String, Box<dyn Reflect>)> for DynamicStruct {
```

Comment:
These are cool! I've been wanting a succinct way to create `DynamicStruct` and friends for a long time haha